### PR TITLE
[WIP] [PyTorch] [STFT] use fft when fft_size == frame_length

### PR DESCRIPTION
**Pull request by ssnl**
Addresses #7883 
Using script in #7883 , here are the new timings:
```
librosa: 1.918302297592163
(2049, 2647)
torch: 0.34431028366088867
torch.Size([10, 2639, 2049])
```
The output shape difference is expected as explained in https://github.com/pytorch/pytorch/issues/7038 .

### CODE REVIEW

**Comment by SsnL** [comment]
cc @zou3519 for another interesting failing instance of `test_expect` on mac:
```
19:32:09 - [pip] torch (0.5.0a0+23384)
19:32:09 ?                     ------
19:32:09 + [pip] torch (0.5.0a0)
```

File: aten/src/ATen/native/SpectralOps.cpp [hunk 1]
```diff
@@ -216,12 +217,12 @@ Tensor stft(const Tensor& self, const int64_t frame_length,
   }
   if (hop <= 0) {
     std::ostringstream ss;
-    REPR(ss) << " expected hop > 0, but got hop=" << hop;
+    REPR(ss) << ": expected hop > 0, but got hop=" << hop;
```
Comment (ezyang) [inline aten/src/ATen/native/SpectralOps.cpp]: FYI you can conveniently do this with `AT_CHECK` now (but you would have to rewrite `REPR` macro, so maybe not for this patch)

**Review by ezyang — APPROVED** [review]

**Review by ezyang — CHANGES_REQUESTED** [review]
reconciling with librosa

**Comment by SsnL** [comment]
i'll re-submit the pr after #8721

**Pull request closed without merging**